injector/__init__.py

```diff
@@ -1367,6 +1367,23 @@ def provider(function: CallableT) -> CallableT:
     return function
 
 
+def named_provider(name: str) -> CallableT:
+    """Decorator for :class:`Module` methods, creating annoted type a of a type.
+
+    >>> class MyModule(Module):
+    ...   @named_provider("first")
+    ...   def provide_name(self) -> str:
+    ...       return 'Bob'
+
+    """
+
+    def decorator(function: CallableT):
+        _mark_named_provider_function(function, name, allow_multi=False)
+        return function
+
+    return decorator
```

Comment:
I'm not convinced that this alternative way of declaring a provider is worth the extra maintenance it introduces. Do you have any arguments behind introducing it?